### PR TITLE
feat: implement zip combinators functions in array

### PR DESCRIPTION
**Pull request by OverGlass**
implement + test
- zip
- zipWith
- unzip

### CODE REVIEW

File: array.ts [hunk 1]
```diff
@@ -248,6 +248,48 @@ export const deleteAt = (i: number) =>
...
+): ReadonlyArray<[A, B]>;
+export function zip<A, B>(
+  as: ReadonlyArray<A>,
+  bs?: ReadonlyArray<B>,
+): ReadonlyArray<[A, B]> | ((bs: ReadonlyArray<B>) => ReadonlyArray<[B, A]>) {
+  if (bs === undefined) {
+    return (bs) => zip(bs, as);
+  }
+  return zipWith(as, bs, (a, b) => [a, b]);
+}
```
Comment (baetheus) [inline array.ts:279]: For now, I'd prefer we not overload any function implementations based around auto-curry. For this particular function I wouldn't curry it at all.

**Review by baetheus — CHANGES_REQUESTED** [review]
Looks good! Aside from pulling out the auto-curry I'd like to see a few more tests around common edge conditions such as empty arrays, large arrays (10k plus items), and larger/smaller sized arrays on both inputs for the zip functions.

File: array.ts (hunk 1 above)
Comment (OverGlass) [reply array.ts]: Hello !
I removed overload and add tests for zip, but it still curried. (Tell me if you really want it without currying)
To do large array test I implemented range function.
I refactored zipWith like fp/ts 3.0.0
